tests/integration_tests/performance/test_net_tput.py

```diff
@@ -0,0 +1,322 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests the network throughput overhead added by Firecracker."""
+
+import json
+import time
+import statistics
+import socket
+from subprocess import run, PIPE
+import pytest
+
+from retry import retry
+
+import host_tools.network as net_tools
+from framework.statistics import Statistics
+
+IPERF = 'iperf3'
+OUTPUT_JSON = 'test_net_tput_{}_iperfs.json'
+CSET_NAME = 'test_tput'
+
+class IperfClient:
+    # Flow direction and corresponding modes for iperf.
+    MODES = {
+        'h2g': '-R',
+        'g2h': ''
+    }
+    # Iteration duration of a single run in seconds.
+    ITERATION_DURATION = 1
+    # Number of test runs.
+    RUNS = 1
+    PINNING_CPU = 1
+
+    def __init__(self, protocol, host_ip, iperfs, microvm):
+        assert protocol == 'udp' or protocol == 'tcp'
+        self.protocol = protocol
+        self.host_ip = host_ip
+        self.iperfs = iperfs
+        self.microvm = microvm
+
+    def build_params_map(self):
+        """Build parameters map used for iperf."""
+        specific_params = self.get_specific_params()
+        params = {
+            'host_ip': self.host_ip,
+            'iperfs': self.iperfs,
+            'protocol': specific_params['protocol'],
+        }
+
+        for _ in range(self.RUNS):
+            for bw in specific_params['bandwidth']:
+                for mode in self.MODES.items():
+                    params['bw'] = bw
+                    params['mode'] = mode[1]
+
+                    yield params
+
+    def build_iperf_cmd(self, params):
+        """Build the iperf command applying given params."""
+        self.ssh_connection = net_tools.SSHConnection(self.microvm.ssh_config)
+
+        cmd = '{} -c {} -t{} -f MBytes -J -b {}M {} {} -A {} -P {}'.format(
```

Comment:
There are a lot of hardcoded strings here, and the logic seems quite hard to read. I was wondering if it would make sense to have a builder class here and do something like:

```
cmd = iperf_cmd_builder.client(ip).iteration_duration(self.ITERATION_DURATION).bandwidth(params['bw'])....build()
```
Each function call would append to a string. For example `iperf_cmd_builder.client(ip)` would do something like `self.cmd += '-c ' + ip`. And build() would return the final command.

Maybe this would also help us remove the `build_params_map()`, since we could use the `iper_cmd_builder` as a temporary storage for the incomplete command, instead of the `params_map`